pkg/distributor/otel.go

```diff
@@ -260,9 +260,8 @@ func otlpHandler(
 				errorMsg string
 			)
 			if st, ok := grpcutil.ErrorToStatus(err); ok {
-				// TODO: This code is needed for backwards compatibility,
-				// and can be removed once -ingester.return-only-grpc-errors
-				// is removed.
+				// This code is needed for a correct handling of errors returned by the supplier function.
+				// These errors are created by using the httpgrpc package.
```

Comment:
can we just return a `newValidationError` from the supplier and delete this piece of code too?

Reply:
Hi @dimitarvdimitrov 
Errors can also come from distributors, and they are never built by the `httprgrpc` package.

Reply:
I'm not sure I follow. The comment now says `This code is needed for a correct handling of errors returned by the supplier function.`. Can we change the supplier function to return non-httpgrpc errors so that we can remove the `if` branch? My understanding is that then there will be no piece of code within the distributor which returns an `httprgrpc` error

Reply:
We can decide to change the current implementation and to remove usage of the `httpgrpc` package from the `distributor`package, but that might not be an easy task, since the full parsing of requests both in otlp and http handler have to be changed. I am not even sure that we want to do it, because the otlp handler is supposed to return `httpgrpc` errors (to be verified).
However, that is not the goal of this PR, which removes the legacy code remained after `-ingester.return-only-grpc-errors` was removed.

Reply:
Hi @dimitarvdimitrov, 
WDYT about my previous comment?

Reply:
ah, ok, then I misunderstood what the comment above said. By `supplier` you actually mean the `supplier` and the `praser` together, which vastly increases the scope. 

Then I'm happy with the changes you've made